kafka-ui-react-app/src/redux/reducers/schemas/reducer.ts

```diff
@@ -0,0 +1,44 @@
+import { SchemaSubject } from 'generated-sources';
+import ActionType from 'redux/actionType';
+import { Action, SchemasState } from 'redux/interfaces';
+
+export const initialState: SchemasState = {
+  byName: {},
+  allNames: [],
+};
+
+const updateSchemaList = (
```

Comment:
I would suggest doing all manipulation with data in thunks and in reducer just apply it.